lib/riffed/client.ex

```diff
@@ -207,12 +223,17 @@ defmodule Riffed.Client do
           {:error, :closed} ->
             new_client = Client.reconnect(client)
             call_thrift(new_client, call_name, args, retry_count + 1)
-          err = {:error, _} ->
-            {new_client, err}
+
+          err = {:error, reason} ->
+            # try to close this, this process is going to die anyway.
```

Comment:
I follow the reasoning. The one thing I'm still fuzzy on is which cases close `thrift_client` here (on the next line) and which cases close the client through the trapped exit path.